facet-reflect/src/peek/list.rs

```diff
@@ -1,24 +1,38 @@
 use super::Peek;
-use core::fmt::Debug;
-use facet_core::ListDef;
+use core::{fmt::Debug, marker::PhantomData};
+use facet_core::{ListDef, PtrConst, PtrMut};
 
 /// Iterator over a `PeekList`
 pub struct PeekListIter<'mem, 'facet_lifetime> {
-    list: PeekList<'mem, 'facet_lifetime>,
+    state: PeekListIterState<'mem>,
     index: usize,
     len: usize,
+    def: ListDef,
+    _list: PhantomData<Peek<'mem, 'facet_lifetime>>,
```

Comment:
I'm fuzzy on what `'facet_lifetime` is for, but I _think_ this `PhantomData` gives `PeekListIter` the same variance that had before this PR?

Reply:
We allow exactly one lifetime for types that derive `Facet`, and it is invariant. That's what `facet_lifetime` is. I think we should call it `borrow` or something, but it's hard to name correctly :(

Reply:
I opened https://github.com/facet-rs/facet/issues/636 to track the naming issue btw